### PR TITLE
Populate fallback fonts from CIPD in tests

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This fixes https://github.com/flutter/flutter/issues/85793

The script that generates the font fallback data also rolls a package to CIPD with these files and updates the DEPS to download those files from CIPD. This makes sure that these tests are hermetic.

### CODE REVIEW

**Review by mdebbar — APPROVED** [review]
Thanks for refactoring CIPD utilities too!

**Review by zanderso — COMMENTED** [review]
Would you mind throwing some documentation under https://github.com/flutter/engine/tree/main/tools/cipd explaining how the CIPD package is created and uploaded? Thanks!

File: lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart [hunk 1]
```diff
@@ -5,21 +5,23 @@
...
-import 'noto_font.dart';
+import 'package:ui/src/engine.dart';
 
 /// Global static font fallback data.
 class FontFallbackData {
+
+  factory FontFallbackData() =>
+    FontFallbackData._(getFallbackFontData(configuration.useColorEmoji));
+
+  FontFallbackData._(List<NotoFont> fallbackFonts) :
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart:16]:
```suggestion
  FontFallbackData._(this.fallbackFonts) :
```

File: lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart [hunk 2]
```diff
@@ -39,10 +41,20 @@ class FontFallbackData {
   /// Code units which are known to be covered by at least one fallback font.
   final Set<int> knownCoveredCodeUnits = <int>{};
 
+  final List<NotoFont> fallbackFonts = getFallbackFontData(configuration.useColorEmoji);
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart:44]:
```suggestion
  final List<NotoFont> fallbackFonts;
```

**Review by zanderso — APPROVED** [review]

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart (hunk 2 above)
Comment (eyebrowsoffire) [reply lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart]: Good catch! Thanks.

File: lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart [hunk 3]
```diff
@@ -430,7 +428,10 @@ void testMain() {
         }
       }
     });
-  }, skip: isSafari);
+  },
+  skip: isSafari,
+  // These tests can take a long time, particularly when compiled to wasm.
+  timeout: const Timeout.factor(4));
```
Comment (Hixie) [inline lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart:434]: changing the timeout violates our style guide

File: DEPS [hunk 4]
```diff
@@ -913,6 +913,16 @@ deps = {
     ],
     'condition': 'download_impeller_cmake_example and host_os == "mac"',
     'dep_type': 'cipd',
+  },
+
+  'src/third_party/flutter_font_fallbacks': {
```
Comment (Hixie) [inline DEPS:918]: we might want to call these something that more clearly indicates their provenance (Google Fonts) and use (testing only), e.g. `google_fonts_for_unit_tests` or something

**Pull request merged**